### ISSUE TITLE
CVC4 produces a satisfiable unsat-core on QF_S formula

### ISSUE THREAD

**Issue by muchang** [cvc5/cvc5#5604]
For this formula, CVC4 produces a satisfiable unsat-core:
```
[569] % cvc4 --strings-exp --check-unsat-cores -q small.smt2
Fatal failure within void CVC4::SmtEngine::checkUnsatCore() at src/smt/smt_engine.cpp:1464
Internal error detectedSmtEngine::checkUnsatCore(): produced core was satisfiable.
Aborted
[570] % 
[570] % cat small.smt2
(declare-fun a () String)
(declare-fun b () String)
(declare-fun c () String)
(assert (xor (str.prefixof (str.replace "B" (str.substr a 0 (str.len b)) "") b) (str.prefixof "B" b)))
(assert (= a (str.++ b c)))
(check-sat)
[571] %
```

Commit: a3395c1

**Comment by HanielB** [comment]
As an internal note, this works on proof-new. When we start using it for unsat-cores we will close this issue.

**Comment by HanielB** [comment]
Closed via #6574
